addons/web/static/src/views/list/list_renderer.js

```diff
@@ -626,6 +629,14 @@ export class ListRenderer extends Component {
         return Object.values(propertyGroups);
     }
 
+    /**
+     * This method is added so that overrides of ListRenderer are able to not display
+     * some columns in the field visibility drop-down menu.
+     */
+    getIgnoredColumns() {
```

Comment:
Sorry but I don't like this either. I need to understand why you need it. The logic here is already complex, we don't want to complexify it even more with visible fields but that should be ignored. This doesn't make any sense. Why `invisible="1"` isn't enough?

Reply:
I can't find where you override this, so the diff in this file might no longer be necessary